test/cql-pytest/test_in_operator.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+from cassandra import InvalidRequest
+from cassandra.cluster import Session, NoHostAvailable
+from cassandra.protocol import SyntaxException
+
+from util import unique_name
+
+
+@pytest.fixture(scope="module")
+def table_name(request, cql: Session, test_keyspace: str) -> str:
+    table = f"{test_keyspace}.{unique_name()}"
+
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, a ascii, b boolean, d double,"
+                f"mai map<ascii, int>, tup frozen<tuple<text, int>>, l list<text>)")
+    request.addfinalizer(lambda: cql.execute("DROP TABLE " + table))
+    return table
+
+
+@pytest.mark.parametrize("column,column_type", [
+    ("mai", "map<ascii, int>"),
+    ("l", "list<text>"),
+])
+def test_in_query_with_invalid_column_type(cql: Session, table_name: str, column: str, column_type: str):
+    expected_error = 'Error from server: code=2200 [Invalid query] message="' \
+                     f'Collection column \'{column}\' ({column_type}) cannot be restricted by a \'IN\' relation"'
+    with pytest.raises(InvalidRequest) as error:
+        cql.prepare(f"SELECT * FROM {table_name} where {column} IN (?, ?) ALLOW FILTERING;")
+    assert error.value.args[0] == expected_error
```

Comment:
Instead of the separate raises() and assert of the error's content, you can do it in one call - 
    with pytest.raises(InvalidRequest, match='some error message'):
It's shorter and (I think) nicer.
Also, please verify that this test passes also in Cassandra (the script test/cql-pytest/run-cassandra canl run Cassandra for you and run the test against it). I'm worried that the error message might not be character-to-character identical, so the test would fail. I usually try to pick only a part of the error message (just enough to make sure a random syntax error isn't mistaken for the expected error instead of such long error message.